GitUpKit/Views/GIQuickViewController.m

```diff
@@ -235,3 +251,212 @@ - (BOOL)diffFilesViewController:(GIDiffFilesViewController*)controller handleKey
 }
 
 @end
+
+#import "GICommitListViewController.h"
+@interface GIQuickViewControllerWithCommitsList () <GICommitListViewControllerDelegate>
+@property (strong, nonatomic, readonly) GICommitListViewController *leftController;
+@property (strong, nonatomic, readonly) GIQuickViewController *rightController;
+
+@property (strong, nonatomic, readwrite) NSLayoutConstraint *hiddenConstraint;
+@property (strong, nonatomic, readwrite) NSLayoutConstraint *revealedConstraint;
+@end
+
+@implementation GIQuickViewControllerWithCommitsList
+
+#pragma mark - Accessors
+- (GICommitListViewController *)leftController {
+  return self.childViewControllers.firstObject;
+}
+
+- (GIQuickViewController *)rightController {
+  return self.childViewControllers.lastObject;
+}
+
+#pragma mark - Check
+- (BOOL)isHistoryShown {
+  return self.leftController.results.count > 0;
+}
+
+#pragma mark - Actions
+- (void)toggleLeftView {
+  BOOL shouldReveal = self.isHistoryShown;
+  self.hiddenConstraint.active = !shouldReveal;
+  self.revealedConstraint.active = shouldReveal;
+  [self.view setNeedsLayout:YES];
+//  [self.view setNeedsDisplay:YES];
+  [NSAnimationContext runAnimationGroup:^(NSAnimationContext * _Nonnull context) {
+    context.duration = 0.25;
+    context.allowsImplicitAnimation = YES;
+    [self.view layout];
+//    [self.view displayIfNeeded];
```

Comment:
You should never explicitly call `layout`. We should be using `self.view.needsDisplay = YES;` Also get rid of commented out code.

Reply:
Should I use `-layoutSubtreeIfNeeded`?